packages/base-driver/lib/basedriver/capabilities.js

```diff
@@ -334,77 +334,85 @@ function processCapabilities(
 }
 
 /**
- * Return a copy of a capabilities object which has taken everything within the 'options'
- * capability and promoted it to the top level.
+ * Return a copy of a "bare" (single-level, non-W3C) capabilities object which has taken everything
+ * within the 'appium:options' capability and promoted it to the top level.
  *
  * @template {Constraints} C
- * @param {import('@appium/types').W3CCapabilities<C>} originalCaps
- * @return {import('@appium/types').W3CCapabilities<C>} the capabilities with 'options' promoted if necessary
+ * @param {NSCapabilities<C>} obj
```

Comment:
I am not sure this is quite right.  I think what it expects is something more like:

```ts
NSCapabilities<C> & {'appium.options'?: Record<string,any>}
```

because NSCapabilities does not know anything about `appium:options` (I remember we discussed this).  If you do this, on L381 you won't need a type assertion to `string`, since it will be known that the keys are strings.